controllers/platformnetwork_controller.go

```diff
@@ -1325,3 +1333,51 @@ func (r *PlatformNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PlatformNetwork{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PlatformNetworkReconciler) checkRestoreInProgress(instance *starlingxv1.PlatformNetwork) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annoatation
+func (r *PlatformNetworkReconciler) RestorePlatformNetworkStatus(instance *starlingxv1.PlatformNetwork) error {
+	r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Status without reconcilation: %t", instance.Status.InSync)
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	if annotation != nil {
+		config, ok := annotation[cloudManager.RestoreInProgress]
+		if ok {
+			restoreStatus := &cloudManager.RestoreStatus{}
+			err := json.Unmarshal([]byte(config), &restoreStatus)
+			if err == nil {
+				if restoreStatus.InSync != nil {
+					instance.Status.InSync = *restoreStatus.InSync
+				}
+				if restoreStatus.Reconciled != nil {
+					instance.Status.Reconciled = *restoreStatus.Reconciled
+				}
+				if restoreStatus.DeploymentScope != nil &&
+					(strings.ToLower(*restoreStatus.DeploymentScope) == "bootstrap" ||
+						strings.ToLower(*restoreStatus.DeploymentScope) == "principal") {
```

Comment:
What if restoreStatus.Reconciled is nil? The pointer would fail to resolve and panic.

Check to determine if restoreStatus.DeploymentScope is one of ("bootstrap", "principal") should be in nested conditional block.

Reply:
Changes are done